### PR TITLE
Enhance .gitignore, improve ci steps

### PR DESCRIPTION
**Pull request by gostkin**
idea files are ignored

added commented fmt and clippy steps

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+*.swp
+
+# IDE
+.idea/
+.vscode/
+
+# Rust
+target/
+book/
+Cargo.lock
```
Comment (rooooooooob) [inline .gitignore:11]: I thought this should only be ignored for libraries, not applications?

**Comment by SimonIT** [comment]
Is there anything missing to get merged or what is the state?

**Comment by rooooooooob** [comment]
@SimonIT Far too many conflicts to merge this. We should wait until all other open PRs are merged before attempting rustfmt again.

**Comment by SebastienGllmt** [comment]
Depending on what we want to do with #112 , now may be a good time to reattempt rustfmt

**Pull request closed without merging**